### PR TITLE
Updated Cipher.crysl to support updateAAD method

### LINKED ISSUE #9: Cipher rule misses updateAAD events
A Cipher object has methods `updatesAAD ` which update the Additional Authentication Data (AAD). The Cipher rule misses these events.

### PR DESCRIPTION
**Pull request by svenfeld**
fixes #9 
This pull requests adds the necessary objects to support the updateAAD method of the [Cipher](https://docs.oracle.com/javase/7/docs/api/javax/crypto/Cipher.html) class. To restrict the usage of the updateAAD method to only the GCM/CCM mode of operation there is a CONSTRAINT added that forbids a call to updateAAD in other modes of operations.

### CODE REVIEW

File: JavaCryptographicArchitecture/src/Cipher.crysl [hunk 1]
```diff
@@ -27,6 +29,9 @@ OBJECTS
 	java.nio.ByteBuffer plainBuffer;
 	java.nio.ByteBuffer cipherBuffer;
 	
+	byte[] aad;
```
Comment (enriozuni) [inline JavaCryptographicArchitecture/src/Cipher.crysl:32]: I would suggest a better naming for this, like `aad_byte`.

File: JavaCryptographicArchitecture/src/Cipher.crysl [hunk 2]
```diff
@@ -27,6 +29,9 @@ OBJECTS
...
+	java.nio.ByteBuffer aadBuffer;
```
Comment (enriozuni) [inline JavaCryptographicArchitecture/src/Cipher.crysl:33]: Same as before, I would suggest a better naming here as well, like `aad_bytebuffer`.

**Review by enriozuni — CHANGES_REQUESTED** [review]
Everything looks good, apart from the two minor refactorings that I have suggested.

File: JavaCryptographicArchitecture/src/Cipher.crysl (hunk 1 above)
Comment (svenfeld) [reply JavaCryptographicArchitecture/src/Cipher.crysl]: I changed it to your suggestion.

File: JavaCryptographicArchitecture/src/Cipher.crysl (hunk 2 above)
Comment (svenfeld) [reply JavaCryptographicArchitecture/src/Cipher.crysl]: I changed it to your suggestion and furthermore changed it for the plaintext and ciphertext bytebuffer aswell to have it consistent.

**Review by enriozuni — APPROVED** [review]
Everything looks good :)

**Pull request merged**